wincent/liferay-portal
# LPS-112897 Update modal usages in journal

Hey @wincent ,

In this PR, we are updating modal usages in journal modules. 

Notes:
- One usage will remain, [this one here](https://github.com/liferay/liferay-portal/blob/5e5d995af581609ea5cb5cbcd3ae57b473317616/modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/ManagementToolbarDefaultEventHandler.es.js#L69). It has a unique complication, explained in https://issues.liferay.com/browse/LPS-113200
- I added change to always close modal after select. We are doing this in `selectEntityHandler`, [here](https://github.com/liferay/liferay-portal/blob/5e5d995af581609ea5cb5cbcd3ae57b473317616/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js#L1342), but this is not applicable to new modals. This is pretty opinionated behavior, but it matches what we have. If there is a need, we could consider extending API with something like `closeOnSelect` boolean.
- I renamed `onSelect` argument to `selectedItem`, where it wasn't named that already. I find `event` to be misleading, as this is a plain object unrelated to JS event object. 

There are many usages to test :) 

1. Open Control Panel > Site > Content & Data > Web Content
2. Create new web content (+), publish
3. Web content item options > Permissions 
-
1. Web content item options > Preview
-
1. Edit web content
1. Sidebar > Default Template > Eye Icon
1. Select a template or no template, confirm
-
1. Update web content few times, changing text a bit
1. Web content item options > View History
1. Version item options > Compare To
1. Select a version
-
1. Edit web content
1. Open options on top right, next to user icon > View Source
-
1. Templates tab > create a new template (+)
1. In the new template creation screen, assign 'Basic Web Content' as a Structure.
1. Save, for later tests.
-
1. Templates tab > template item options > Permissions
-
1. Web Content tab > Edit web content
1. Sidebar > Default Template > Select
1. Select newly created template
-
1. Web Content tab > create a new folder (+)
1. Folder item options > Edit 
1. Structure Restrictions and Workflow > Define Specific > Choose Structure
-
1. Web Content tab > Filter and Order dropdown > Structures
-
1. Structures tab > create a new structure (+)
1. Properties > Select a Parent Structure
-
1. Put a web content display widget on a page.
1. WCD app options > Configuration > select a web content
1. Template section > Use a specific template > Select
-
1. Control Panel > Configuration > System Settings > Web Content > System Scope > Administration. Check 'Show Feeds'
1. There should be a new tab on Web Content page named `Feeds`.
1. Feeds tab > create a new feed (+)
1. Web Content Constraints > Select a Structure

Please review the code.

Thank you!

/cc @jbalsas
